### PR TITLE
docs: add demo.tutorialkit.dev website

### PR DESCRIPTION
**Pull request by SamVerschueren**
This PR brings the source of https://demo.tutorialkit.dev to this repository to make it easier for external contributors to contribute.

I updated the dependencies to use the workspace dependencies instead. This means that the demo is now always build with the version of TK that is on `main`.

### CODE REVIEW

File: docs/demo/package.json [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+    "astro": "astro",
+    "build_with_check": "astro check && astro build",
+    "build": "astro build",
+    "dev": "astro dev",
+    "preview": "astro preview",
+    "start": "astro dev",
+    "postbuild": "cp _headers ./dist/"
+  },
+  "dependencies": {
+    "@tutorialkit/components-react": "0.0.1-alpha.22",
```
Comment (SamVerschueren) [inline docs/demo/package.json:16]: I will switch to using the `workspace` packages instead. But first I need this https://github.com/stackblitz/tutorialkit/pull/84 to land.

File: docs/demo/README.md
[the remark this answers is not in the document]
Comment (SamVerschueren) [reply docs/demo/README.md:1]: No I'd rather keep it documented here. Angular also does it. I don't want to bloat the root readme with tons of sections. If people want to contribute to the demo (or the docs), they can just open that folder and read how to start it.

**Review by SamVerschueren — COMMENTED** [review]
Great work on going over all of the new changes @Nemikolh  🙏!

File: docs/demo/src/content/tutorial/meta.md [hunk 2]
```diff
@@ -0,0 +1,7 @@
+---
+type: tutorial
+title: Basics
```
Comment (Nemikolh) [inline docs/demo/src/content/tutorial/meta.md:3]: Oh interesting, we also have that property in `overrides/src/content/tutorial/meta.md`. We should remove it from there.

```suggestion
```

**Pull request merged**